pkg/integration/replication/synchronous_replication_test.go

```diff
@@ -70,6 +72,61 @@ func (suite *SyncTestSuite) TestMasterRestart() {
 	}
 }
 
+// TestPrecommitStateSync checks if the precommit state at master
+// and its followers are in sync during synchronous replication
+func (suite *SyncTestSuite) TestPrecommitStateSync() {
+	var (
+		masterState *schema.ImmutableState
+		err         error
+		startCh     = make(chan bool)
+	)
+
+	ctx, client, cleanup := suite.ClientForMaser()
+	defer cleanup()
+
+	// Create goroutines for client waiting to query the state
+	// of the followers. This is initialised before to avoid
+	// spending time initialising the follower client for faster
+	// state access
+	var wg sync.WaitGroup
+	for i := 0; i < suite.GetFollowersCount(); i++ {
+		wg.Add(1)
+		go func(followerID int) {
+			defer wg.Done()
+			ctx, client, cleanup := suite.ClientForReplica(followerID)
+			defer cleanup()
+			for {
+				select {
+				case <-startCh:
+					suite.Run(fmt.Sprintf("test replica sync state %d", followerID), func() {
+						state, err := client.CurrentState(ctx)
+						require.NoError(suite.T(), err)
+						suite.Require().Equal(state.PrecommittedTxId, masterState.PrecommittedTxId)
+						suite.Require().Equal(state.PrecommittedTxHash, masterState.PrecommittedTxHash)
```

Comment:
Shouldn't we check against Committed state on master ?

```suggestion
						suite.Require().Equal(state.PrecommittedTxId, masterState.TxId)
						suite.Require().Equal(state.PrecommittedTxHash, masterState.TxHash)
```

Actually at that point on the primary the Precomitted and Comitted should be the same (since we don't have any pending unconfirmed transaction) - we could check that right after getting `CurrentState` from primary.

Reply:
ack, makes sense